torchmetrics/functional/regression/tweedie_deviance.py

```diff
@@ -0,0 +1,147 @@
+
+    Args:
+        preds: Predicted tensor
+        targets: Ground truth tensor
+        power:
+            - power < 0 : Extreme stable distribution. (Requires: preds > 0.)
+            - power = 0 : Normal distribution. (Requires: targets and preds can be any real numbers.)
+            - power = 1 : Poisson distribution. (Requires: targets >= 0 and y_pred > 0.)
+            - 1 < p < 2 : Compound Poisson distribution. (Requires: targets >= 0 and preds > 0.)
+            - power = 2 : Gamma distribution. (Requires: targets > 0 and preds > 0.)
+            - power = 3 : Inverse Gaussian distribution. (Requires: targets > 0 and preds > 0.)
+            - otherwise : Positive stable distribution. (Requires: targets > 0 and preds > 0.)
+
+    Example:
+        >>> targets = torch.tensor([1.0, 2.0, 3.0, 4.0])
+        >>> preds = torch.tensor([4.0, 3.0, 2.0, 1.0])
+        >>> _tweedie_deviance_score_update(preds, targets, power=0)
+        (tensor(20.), tensor(4))
+    """
+    _check_same_shape(preds, targets)
+
+    zero_tensor = torch.zeros(preds.shape, device=preds.device)
+
+    if 0 < power < 1:
+        raise ValueError(f"Deviance Score is not defined for power={power}.")
+
+    if power == 0:
+        deviance_score = torch.pow(targets - preds, exponent=2)
+    elif power == 1:
+        # Poisson distribution
+        if torch.any(preds <= 0) or torch.any(targets < 0):
+            raise ValueError(
+                f"For power={power}, 'preds' has to be strictly positive and 'targets' cannot be negative."
+            )
+
+        deviance_score = 2 * (targets * torch.log(targets / preds) + preds - targets)
+    elif power == 2:
+        # Gamma distribution
+        if torch.any(preds <= 0) or torch.any(targets <= 0):
+            raise ValueError(f"For power={power}, both 'preds' and 'targets' have to be strictly positive.")
+
+        deviance_score = 2 * (torch.log(preds / targets) + (targets / preds) - 1)
+    else:
+        if power < 0:
+            if torch.any(preds <= 0):
+                raise ValueError(f"For power={power}, 'preds' has to be strictly positive.")
+        elif 1 < power < 2:
+            if torch.any(preds <= 0) or torch.any(targets < 0):
+                raise ValueError(
+                    f"For power={power}, 'targets' has to be strictly positive and 'preds' cannot be negative."
+                )
+        else:
+            if torch.any(preds <= 0) or torch.any(targets <= 0):
+                raise ValueError(f"For power={power}, both 'preds' and 'targets' have to be strictly positive.")
+
+        term_1 = torch.pow(torch.max(targets, zero_tensor), 2 - power) / ((1 - power) * (2 - power))
+        term_2 = targets * torch.pow(preds, 1 - power) / (1 - power)
+        term_3 = torch.pow(preds, 2 - power) / (2 - power)
+        deviance_score = 2 * (term_1 - term_2 + term_3)
+
+    sum_deviance_score = torch.sum(deviance_score)
+    num_observations = torch.tensor(torch.numel(deviance_score))
```

Comment:
device mismatch possible
```suggestion
    num_observations = torch.tensor(torch.numel(deviance_score), device=preds.device)
```